lib/pbench/server/auth/auth.py

```diff
@@ -138,26 +198,42 @@ def verify_third_party_token(auth_token: str, oidc_client: OpenIDClient) -> bool
         Returns:
             True if the verification succeeds else False
         """
-        identity_provider_pubkey = oidc_client.get_oidc_public_key(auth_token)
         try:
-            oidc_client.token_introspect_offline(
+            identity_provider_pubkey = oidc_client.get_oidc_public_key(auth_token)
+        except Exception:
+            Auth.logger.warning("Identity provider public key fetch failed")
+            identity_provider_pubkey = Auth().get_secret_key()
+        try:
+            audience = oidc_client.client_id if oidc_client else None
+            token_decode = oidc_client.token_introspect_offline(
                 token=auth_token,
                 key=identity_provider_pubkey,
-                audience=oidc_client.client_id,
+                audience=audience,
+                algorithms=algorithms,
                 options={
                     "verify_signature": True,
                     "verify_aud": True,
                     "verify_exp": True,
                 },
             )
-            return True
+            roles = []
+            if audience in token_decode.get("resource_access"):
+                roles = token_decode.get("resource_access").get(audience).get("roles")
+            return InternalUser(
+                id=token_decode.get("sub"),
+                username=token_decode.get("preferred_username"),
+                email=token_decode.get("email"),
+                first_name=token_decode.get("given_name"),
+                last_name=token_decode.get("family_name"),
+                roles=roles,
+            )
         except (
             jwt.ExpiredSignatureError,
             jwt.InvalidTokenError,
             jwt.InvalidAudienceError,
         ):
             Auth.logger.error("OIDC token verification failed")
```

Comment:
These conditions sound like user errors, not problems with the server. If we're going to log them at all, seems they shouldn't likely be more severe than `warning`??